packages/pwa-kit-dev/bin/pwa-kit-dev.js

```diff
@@ -143,46 +146,76 @@ const main = async () => {
+                    })
+                    .env('MRT_API_KEY')
+                    .conflicts('credentialsFile')
             )
             .hook('preAction', (thisCommand, actionCommand) => {
+                let {cloudOrigin, credentialsFile, user, key} = actionCommand.opts()
+
+                // support older PWA_KIT_CREDENTIALS_FILE env var for credentialsFile
+                if (!credentialsFile && process.env.PWA_KIT_CREDENTIALS_FILE) {
+                    credentialsFile = process.env.PWA_KIT_CREDENTIALS_FILE
+                }
+
                 // The final credentialsFile path depends on both cloudOrigin and credentialsFile opts.
                 // Pre-process before passing to the command.
-                const {cloudOrigin, credentialsFile} = actionCommand.opts()
                 actionCommand.setOptionValue(
                     'credentialsFile',
                     scriptUtils.getCredentialsFile(cloudOrigin, credentialsFile)
                 )
+
+                // user and key should always be used together
+                if ((user && !key) || (!user && key)) {
+                    throw new program.InvalidArgumentError(
+                        'You must provide a --user and --key option together, or neither'
+                    )
+                }
             })
     }
 
     managedRuntimeCommand('save-credentials')
         .description(`save API credentials for Managed Runtime`)
-        .requiredOption(
-            '-u, --user <email>',
-            'the e-mail address you used to register with Managed Runtime',
-            (val) => {
-                if (!validator.isEmail(val)) {
-                    throw new program.InvalidArgumentError(`"${val}" is not a valid email`)
-                } else {
-                    return val
-                }
-            }
-        )
-        .requiredOption(
-            '-k, --key <api-key>',
-            `find your API key at https://runtime.commercecloud.com/account/settings`,
-            (val) => {
-                if (typeof val !== 'string' || val === '') {
-                    throw new program.InvalidArgumentError(`"api-key" cannot be empty`)
-                } else {
-                    return val
-                }
+        .hook('preAction', (thisCommand, actionCommand) => {
+            const {user, key} = actionCommand.opts()
+            // user and key are optional for other MRT commands but required here as we need something to save
+            if (!user || !key) {
+                throw new program.InvalidArgumentError(
+                    'You must provide a --user and --key option to save to the credentials file'
+                )
             }
-        )
+        })
         .action(async ({user, key, credentialsFile}) => {
             try {
-                fse.writeJson(credentialsFile, {username: user, api_key: key}, {spaces: 4})
+                await fse.writeJson(credentialsFile, {username: user, api_key: key}, {spaces: 4})
```

Comment:
Using `.conflicts` in managedRuntimeCommand` will break `save-credentials` because it means that you can't save credentials to a non-default location. 😞  Could solve this by maybe adding a flag to `managedRuntimeCommand`?

Reply:
This is what I was saving in my comment above: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1499#issuecomment-1781499580

Reply:
I removed calls to `.conflicts`. Username/key will take precedence in push/tail-logs (net new so no breaking change) and save-credentials works as expected with non-default location.